File: src/com/aac/q0198/houserobber/Solution.java
package com.aac.q0198.houserobber;

public class Solution {
    public int rob(int[] nums) {
        if(nums == null)
            return 0;
        int len = nums.length;
        if(len == 0)
            return 0;
        int[] sum = new int[len + 1];
        sum[0] = 0;
        sum[1] = nums[0];
        for (int i = 2; i <= len; ++i) {
            sum[i] = Math.max(sum[i - 1], sum[i - 2] + nums[i - 1]);
        }
        return sum[len];
    }

    public static void main(String[] args) {
        int[] input = {2, 1};
        System.out.println(new Solution().rob(input));
    }
}

File: src/com/aac/q0198/houserobber/Solution.py
from typing import List


class Solution:
    def rob(self, nums: List[int]) -> int:
        dp = [0, 0]
        i = 0
        for num in nums:
            dp[i % 2] = max(num + dp[i % 2], dp[(i + 1) % 2])
            i = i + 1
        return dp[(i + 1) % 2]
